pandas/tests/extension/base/getitem.py

```diff
@@ -134,8 +134,9 @@ def test_take(self, data, na_value, na_cmp):
 
     def test_take_empty(self, data, na_value, na_cmp):
         empty = data[:0]
-        result = empty.take([-1])
-        na_cmp(result[0], na_value)
+        if data._can_hold_na:
+            result = empty.take([-1])
+            na_cmp(result[0], na_value)
```

Comment:
Question is what this should do if `_can_hold_na=False` ? And in general with -1 indexers (eg `reindex`, `concat` with realignment, ...) 
I suppose to raise an error? Or coerce to object?

Reply:
Yes, maybe, but then I am just wondering what this design decision is (and we should also test that here). 

As far as I know, this is rather new terrain (in pandas we only have ints, and they are upcasted to float), so it might be we still need to discuss/decide on this.

Reply:
We also have a BoolBlock that cannot hold NaNs, so a boolean Series gets converted to object Series when NaNs are introduced:

```
In [49]: pd.Series([True, False])
Out[49]: 
0     True
1    False
dtype: bool

In [50]: pd.Series([True, False]).reindex([1, 2])
Out[50]: 
1    False
2      NaN
dtype: object
```

So this is some prior art

Reply:
So by the statement "we still need to discuss/decide on this", I presume that the core dev team will make that call.  Right? (FWIW, I think I may use `_can_hold_na=True`, and since I'm storing objects, I'll just use NaN to be the na_value that is then used in `reindex`, `shift`, etc.), and I'd want the type of the resulting array to remain in my subclass of `ExtensionArray`

Reply:
> So by the statement "we still need to discuss/decide on this", I presume that the core dev team will make that call. Right?

No, this is an open discussion, and your voice in this discussion is important since you are actually one of the first people trying to use this (both Tom and I do not use `_can_hold_na=False` in our applications).

But if you would also not use this, we should maybe consider removing the feature for now (meaning this will always be True) until somebody comes up with a good use case.

Reply:
Well, I've come full circle on this.  In my case, I thought that I would not want NA in my arrays. My objects stored in the arrays have no NA value that makes sense. And then when you brought up `reindex` as an example, I realized that I would use things like `shift` and would want the NA to be filled in, but keeping the `ExtensionArray` class as the holder, meaning that I need `_can_hold_na` as True.  I'd want `ExtensionArray` as holding it so that arithmetic would work, as coercing to an `object` array would mess up the arithmetic. So I'd have no problem if we dropped `_can_hold_na` from the `ExtensionArray` design, and tell people that they must define a _na_value(), which is then used as the "fill in" when we do things like `reindex` and `shift`.  And because there are pandas operations that need some NA value for fill in purposes, requiring it on `ExtensionArray` shouldn't be an issue (IMHO).

If we agree that getting rid of `_can_hold_na` is the thing to do, I'll modify this PR to totally get rid of it (and the associated tests).